lib/net/pool.js

```diff
@@ -2012,6 +2145,147 @@ class Pool extends EventEmitter {
     peer.sendCFCheckpt(packet.filterType, packet.stopHash, filterHeaders);
   }
 
+  /**
+   * Handle peer `CFCheckpt` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {CFCheckptPacket} packet
+   */
+
+  async handleCFCheckpt(peer, packet) {
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+    }
+  }
+
+  /**
+   * Handle peer `CFHeaders` packet.
+   * @method
+   * @private
+   * @param {Peer} peer - Sender.
+   * @param {CFHeadersPacket} packet - Packet to handle.
+   * @returns {void}
+   */
+  async handleCFHeaders(peer, packet) {
+    this.logger.info('Received CFHeaders packet from %s', peer.hostname());
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const filterType = packet.filterType;
+
+    if (filterType !== this.getcfheadersFilterType) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const stopHash = packet.stopHash;
+    assert(stopHash.equals(this.getcfheadersStopHash));
```

Comment:
Let's not use asserts and replace them with bans.

Reply:
Okay. Will replace asserts with bans.